lib/rubocop/packs.rb

```diff
@@ -13,66 +15,38 @@ module Packs
     PACK_LEVEL_RUBOCOP_YML = 'package_rubocop.yml'
     PACK_LEVEL_RUBOCOP_TODO_YML = 'package_rubocop_todo.yml'
 
-    class Error < StandardError; end
-    extend T::Sig
-
-    # Your code goes here...
     PROJECT_ROOT   = T.let(Pathname.new(__dir__).parent.parent.expand_path.freeze, Pathname)
     CONFIG_DEFAULT = T.let(PROJECT_ROOT.join('config', 'default.yml').freeze, Pathname)
     CONFIG         = T.let(YAML.safe_load(CONFIG_DEFAULT.read).freeze, T.untyped)
 
     private_constant(:CONFIG_DEFAULT, :PROJECT_ROOT)
 
-    class Offense < T::Struct
-      extend T::Sig
-
-      const :cop_name, String
-      const :filepath, String
-
-      sig { returns(ParsePackwerk::Package) }
-      def pack
-        ParsePackwerk.package_from_path(filepath)
-      end
-    end
-
-    sig { params(paths: T::Array[String], cop_names: T::Array[String]).returns(T::Array[Offense]) }
-    def self.offenses_for(paths:, cop_names:)
-      path_arguments = paths.join(' ')
-      cop_arguments = cop_names.join(',')
-      # I think we can potentially use `RuboCop::CLI.new(args)` for this to avoid shelling out and starting another process that needs to reload the bundle
-      args = [path_arguments, "--only=#{cop_arguments}", '--format=json']
-      puts "Executing: bundle exec rubocop #{args.join(' ')}"
-      json = JSON.parse(Private.execute_rubocop(args))
-      offenses = T.let([], T::Array[Offense])
-      json['files'].each do |file_hash|
-        filepath = file_hash['path']
-        file_hash['offenses'].each do |offense_hash|
-          offenses << Offense.new(
-            cop_name: offense_hash['cop_name'],
-            filepath: filepath
-          )
-        end
-      end
-
-      offenses
-    end
-
     #
     # Ideally, this is API that is available to us via `rubocop` itself.
     # That is: the ability to preserve the location of `.rubocop_todo.yml` files and associate
     # exclusions with the closest ancestor `.rubocop_todo.yml`
     #
-    sig { params(packs: T::Array[ParsePackwerk::Package]).void }
-    def self.regenerate_todo(packs:)
-      offenses = offenses_for(
-        paths: packs.map(&:name).reject { |name| name == ParsePackwerk::ROOT_PACKAGE_NAME },
+    sig { params(packs: T::Array[ParsePackwerk::Package], files: T::Array[String]).void }
+    def self.regenerate_todo(packs: [], files: [])
```

Comment:
I was a bit torn on this one. Originally, I wanted two separate methods. One that would completely regenerate the TODO from scratch, and another that only adds new offenses to it.

Rubocop **kind of** has something like this:
```
Config Generation:
        --auto-gen-config            Generate a configuration file acting as a
                                     TODO list.
        --regenerate-todo            Regenerate the TODO configuration file using
                                     the last configuration. If there is no existing
                                     TODO file, acts like --auto-gen-config.
```

It felt ultimately like it made more sense that there is one function, `regenerate_todo`, that takes in either packs or files. If you're passing in packs, it completely regenerates it, because it has all the info it needs to regenerate it from scratch. If you're passing in files, it just adds.

I also thought maybe that would result in duplicates if we just add, but then realized we should never have a duplicate (because if it's already there, it shouldn't come in as an offense).

This is slightly different behavior for adding TODOs from individual packs vs from individual files, but it felt like the cleanest trade off.